nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/shape/MergeMaxIndex.java

```diff
@@ -0,0 +1,82 @@
+/* ******************************************************************************
+ * Copyright (c) 2020 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+package org.nd4j.linalg.api.ops.impl.shape;
+
+import lombok.NoArgsConstructor;
+import org.nd4j.autodiff.samediff.SDVariable;
+import org.nd4j.autodiff.samediff.SameDiff;
+import org.nd4j.base.Preconditions;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+import org.nd4j.linalg.api.ops.impl.layers.recurrent.weights.LSTMLayerWeights;
+import org.nd4j.linalg.util.ArrayUtil;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+
+@NoArgsConstructor
+public class MergeMaxIndex extends DynamicCustomOp {
+
+    private DataType dataType = DataType.FLOAT;
+
+    public MergeMaxIndex(SameDiff sameDiff, SDVariable... inputs) {
+        super("mergemaxindex", sameDiff, inputs);
+        addIArgument(this.dataType.ordinal());
+
+    }
+
+    public MergeMaxIndex(INDArray... inputs) {
+        super("mergemaxindex", inputs, null);
+        Preconditions.checkArgument(areEqualShapes(inputs), "All inputs have to be equal shapes");
+        addIArgument(dataType.ordinal());
+
+    }
+
+    public MergeMaxIndex(SameDiff sd, SDVariable[] x, DataType dataType) {
+        super("mergemaxindex", sd, x);
+        addIArgument(dataType.ordinal());
+    }
+
+    public MergeMaxIndex(INDArray[] x, DataType dataType) {
+        super(x, null);
+        Preconditions.checkArgument(areEqualShapes(x), "All inputs have to be equal shapes");
+        addIArgument(dataType.ordinal());
+
+    }
+
+
+    protected boolean areEqualShapes(INDArray... inputs) {
+        for (INDArray input : inputs) {
+            if (!inputs[0].equalShapes(input)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public String opName() {
+        return "mergemaxindex";
+    }
+
+    @Override
+    public List<DataType> calculateOutputDataTypes(List<DataType> dataTypes) {
+        return Collections.singletonList(DataType.INT32);
```

Comment:
Return the datatype - i.e., `private DataType dataType` value - not hardcoded int